src/platforms/gbm-kms/server/kms/display_buffer.cpp

```diff
@@ -63,22 +64,24 @@ mgg::DisplayBuffer::DisplayBuffer(
 {
     listener->report_successful_setup_of_native_resources();
 
-    // TODO: Pull a supported format out of KMS rather than assuming XRGB8888
-    auto initial_fb = std::make_shared<mgg::CPUAddressableFB>(
-        std::move(drm_fd),
-        false,
-        DRMFormat{DRM_FORMAT_XRGB8888},
-        area.size);
-
-    auto mapping = initial_fb->map_writeable();
-    ::memset(mapping->data(), 24, mapping->len());
-
-    visible_fb = std::move(initial_fb);
-    for (auto& output : outputs)
+    if (!smooth_transition)
```

Comment:
So, this is going to break if someone sets `smooth_transition` and the previous resolution was not the same as the resolution our policy has set - we'll have a `view_area()` that doesn't match the actual view area; I'm not sure what else (if anything!) will notice the discrepancy between the display configuration requested and the actual hardware configuration.

It now occurs to me that we could *always* skip setting the CRTC if the current mode is the same as the mode we want to set. The `smooth_boot` option could then be either removed and you get flicker-free boot if and only if the requested resolution is the same as the pre-existing mode, or plumbed through here and a *warning* generated if `smooth_boot` is asked for but cannot be provided because we need to modeset, or handled higher up (there'd need to be some way to pipe the *actual* display configuration up to the display configuration policy? Maybe?).

Reply:
Yeah that sounds like the bug that I would expect! I wonder if there exists some way to query the current resolution and then just set our default view area to that instead?

That's a good point... We would still want the flag for the follow up pull request where we provide a neat transition from the previous screen to the compositor's first frame, but it also makes sense that it would be a sane default.

At any rate, I think we have two options:

1. When we start, we can read the current CRTC and set mir's internals to this resolution. This would avoid _any_ `set_crtc` from happening, as we would always adopt the previous resolution
2. When we start, we can read the current CRTC and compare it to the CRTC that we want. If they mismatch, we would have to `set_crtc` and cause a flicker. Otherwise, we would have a smooth boot.

@Saviq @RAOF thoughts?

Reply:
Note: `drmModeGetCrtc` does what you think it does :smile:

Reply:
Another note: Option 2 is really easy to pull off

Reply:
+1 on doing it by default and doing away with the option altogether :)